### PR TITLE
Standardize middleware to use configureOption lambda

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Standardizing ConfigureOptions to lambda https://github.com/aspnet/Coherence-Signed/issues/144

### CODE REVIEW

File: src/Microsoft.AspNet.Diagnostics/RuntimeInfo/RuntimeInfoExtensions.cs [hunk 1]
```diff
@@ -10,33 +10,45 @@ namespace Microsoft.AspNet.Builder
...
 
-            var libraryManager = builder.ApplicationServices.GetService(typeof(ILibraryManager)) as ILibraryManager;
-            var runtimeEnvironment = builder.ApplicationServices.GetService(typeof(IRuntimeEnvironment)) as IRuntimeEnvironment;
-            return builder.Use(next => new RuntimeInfoMiddleware(next, options, libraryManager, runtimeEnvironment).Invoke);
+            var libraryManager = app.ApplicationServices.GetService(typeof(ILibraryManager)) as ILibraryManager;
+            var runtimeEnvironment = app.ApplicationServices.GetService(typeof(IRuntimeEnvironment)) as IRuntimeEnvironment;
+            var options = new RuntimeInfoPageOptions();
+            configureOptions(options);
+
+            return app.Use(next => new RuntimeInfoMiddleware(next, options, libraryManager, runtimeEnvironment).Invoke);
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Diagnostics/RuntimeInfo/RuntimeInfoExtensions.cs]: This one's weird. It should just be `app.UseMiddleware<RuntimeInfoMiddleware>(options);`. You'll need to re-order the constructor params.

**Comment by Tratcher** [comment]
:shipit:

File: src/Microsoft.AspNet.Diagnostics/RuntimeInfo/RuntimeInfoExtensions.cs (hunk 1 above)
Comment (JunTaoLuo) [inline src/Microsoft.AspNet.Diagnostics/RuntimeInfo/RuntimeInfoExtensions.cs]: RuntimeInfo is currently broken due to #233. I'll make a note of it in that issue to clean this up but I'll merge in the rest now.

**Comment by JunTaoLuo** [comment]
Travis tests are failing due to changes in `TestPlatformHelper`. I'll fix those in a separate commit

**Pull request merged**